### ISSUE TITLE
memory usage during loading of full scan data

### ISSUE THREAD

**Issue by chubukov** [ElucidataInc/ElMaven#69]
We've been using Maven more recently for LC-QTOF data, which is similar to the QExactive data that comprises the majority of the test data so far, with the exception that there is less filtration of low-intensity scans by the instrument. The practical effect is that the produced spectra are very noisy, and the file size is much larger (1GB/sample is not uncommon).

Loading even a medium number (say 30-40 samples) of these into Maven is an enormous memory load. Maven does have some import settings that can filter the data upon loading (options->file import), say, by a minimum intensity. However, something is screwy about the memory usage during this process. I looked at `mzSample::parseMzXMLScan`,`mzSample::addScan`, and `Scan::intensityFilter` and it looks to me like memory should generally be cleared immediately upon filtering each scan (relying on the `vector::swap` trick). But in practice, what happens is that a huge amount of memory is allocated during the file loading, and some of it is cleared only after all the files are loaded. Since memory usage often reaches the limit of available memory during this process, it can really slow down the entire machine.

This may drag us into very low-level memory management issues, but I'd like to see if there are some simple things we can fix to address this.

This is low priority compared to the general instability issues and other November sprint issues -- let's maybe look at it in the next sprint. We'll have to find a way to transfer the files (not sure if dropbox is sufficient).

**Comment by GeorgeSabu** [comment]
@chubukov consuming a large amount of memory for loading samples is something we have observed. I have this problem on the back of my head to solve.

**Comment by chubukov** [comment]
@GeorgeSabu i uploaded some lc-qtof data in 2017_01_09_lcqtof_files on dropbox

**Comment by GeorgeSabu** [comment]
@chubukov This was because of multiprocessing. Because of multiprocessing, more than one file is loaded into the memory to load as multiple files and parsed at the same time.
We have made a checkbox in "File Import" to enable and disable multiprocessing on file upload.
